### PR TITLE
Social Icons - Begin Automating Existing Manual Test Cases.

### PR DESCRIPTION
**Pull request by ttahmouch**
## Description
This is a first pass at beginning to automate some of the manual regression test cases for the Social Icons block types that we assert during some of our release rotations.

They are derived from here:
https://github.com/wordpress-mobile/test-cases/blob/trunk/test-cases/gutenberg/social-icons.md#tc001

The couple tests in this PR cover most of the assertions from this test:
> TC001
> The newly created Social Icons block is added with 4 icons
> Add Social Icons block
> Expect added 4 icons (WordPress, Facebook, Twitter, Instagram)
> Expect WordPress icon to be active, already fulfilled with the link and to have product color

They assert that, by default, WordPress, Facebook, Twitter, and Instagram icons are add added when Social Icons are added. They also assert that WordPress contains a link making it active by default, and the remaining three do not contain a link making them inactive by default. The icon color for the active and inactive icons hasn't been asserted, but the state of activity has been.

I think that when asserting the UX, checking the active state of each of the Social Icons by default makes sense, but I see little value in asserting the color of the icons in each state. That being said, if we want to remain faithful to the original manual test, asserting the colors of the icons would be necessary before the manual test could be removed.

## Testing Instructions

```
npm run native test -- ../block-library/src/social-link/test/index.native.js
```

## Screenshots
N/A

## Types of changes
New "feature." Adding automated integration tests.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

**Comment by fluiddot** [comment]
Hey @ttahmouch 👋, I see you requested a review for this PR but looks like it's still in draft, just wanted to check whether is ready or if you're looking for something specific to get reviewed before marking it as ready for review 🤔 .

**Comment by ttahmouch** [comment]
> Hey @ttahmouch 👋, I see you requested a review for this PR but looks like it's still in draft, just wanted to check whether is ready or if you're looking for something specific to get reviewed before marking it as ready for review 🤔 .

Hey, @fluiddot . I marked it as `Draft` because I still had intended to make some minor refactors, but I ended up dealing with IDE issue with ESLint for a little while yesterday afternoon. Apologies.

I may remove the mocked hook in `test/unit/config/global-mocks.js` since that I don't believe it's necessary with the other mock in `test/native/setup.js`. I also planned to push changes making the default style values for active icons feel less arbitrary in `packages/block-library/src/social-link/edit.native.js`. I was going to also consider if there's any useful utils available or that I could make to simplify staging the tests, but I'm conflicted because I prefer WET instead of DRY test implementations to make them easier to understand.

File: packages/react-native-editor/ios/Podfile.lock [hunk 1]
```diff
@@ -493,12 +493,12 @@ SPEC CHECKSUMS:
   React-logger: faee236598b0f7e1a5e3b68577016ac451f1f993
```
Comment (ttahmouch) [inline packages/react-native-editor/ios/Podfile.lock:493]: These changes were generated. They are unrelated to the implementation details of the component being tested, or the tests.

File: packages/data/src/redux-store/metadata/actions.js [hunk 2]
```diff
@@ -36,9 +36,9 @@ export function finishResolution( selectorName, args ) {
  * Returns an action object used in signalling that a batch of selector resolutions has
```
Comment (ttahmouch) [inline packages/data/src/redux-store/metadata/actions.js:36]: These changes were generated by `npm run lint-js:fix`. They are unrelated to the implementation details of the component being tested, or the tests.

File: packages/compose/src/utils/create-higher-order-component/index.ts [hunk 3]
```diff
@@ -25,8 +25,8 @@ export type HigherOrderComponent< HOCProps extends Record< string, any > > = <
  * Given a function mapping a component to an enhanced component and modifier
```
Comment (ttahmouch) [inline packages/compose/src/utils/create-higher-order-component/index.ts:25]: These changes were generated with `npm run lint-js:fix`. They are unrelated to the implementation details of the component being tested, or the tests.

File: package-lock.json [hunk 4]
```diff
@@ -19364,16 +19364,6 @@
 						"yauzl": "^2.7.0"
 					},
 					"dependencies": {
-						"are-we-there-yet": {
```
Comment (ttahmouch) [inline package-lock.json:19367]: These changes were generated. They are unrelated to the implementation details of the component being tested, or the tests.

File: test/native/setup.js [hunk 5]
```diff
@@ -222,6 +222,12 @@ jest.mock( '@wordpress/compose', () => {
 			mockComponent( 'ResizeObserverMock' )( {} ),
 			{ width: 100, height: 100 },
 		] ),
+		usePreferredColorSchemeStyle: jest.fn( () => {
```
Comment (ttahmouch) [inline test/native/setup.js]: These mock values are effectively "arbitrary," but were taken from values like `$light-quaternary` in `gutenberg/packages/base-styles/_colors.native.scss`

File: packages/block-library/src/social-link/test/index.native.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+		unregisterBlocks();
+	} );
+
+	/**
+	 * GIVEN an EDITOR is displayed;
+	 * WHEN a SOCIAL ICONS BLOCK is selected from the BLOCK INSERTER;
+	 */
+	it( 'should display WORDPRESS, FACEBOOK, TWITTER, INSTAGRAM by default.', async () => {
+		// Arrange
+		const subject = await initializeEditor( {} );
```
Comment (ttahmouch) [inline packages/block-library/src/social-link/test/index.native.js:32]: The staging in each of these tests is purposefully redundant in an effort to keep the code easy to understand in isolation. I don't like DRY code in test suites personally, but the thought crossed my mind about checking for existing utils or creating one. It just doesn't feel worth it at the moment.

File: packages/block-library/src/social-link/test/index.native.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
+/**
```
Comment (ttahmouch) [inline packages/block-library/src/social-link/test/index.native.js:1]: The tests written for the Edit Social Links Block type started to be derived from here:
https://github.com/wordpress-mobile/test-cases/blob/trunk/test-cases/gutenberg/social-icons.md#tc001

They're just rudimentary at the moment to start incrementally, and keep the size of the PR down to easily discuss them. I planned to add more tests in future PRs when I started this.

**Comment by ttahmouch** [comment]
I just noticed some test failures unrelated to the tests in this PR because the snapshot assertions they are running were affected by the default active icon styles I added in the `<EditSocialLink/>` component. I'll try to ensure that the snapshots are updated correctly soon.

https://github.com/WordPress/gutenberg/runs/5324097673?check_suite_focus=true

**Comment by jhnstn** [comment]
@ttahmouch The changes look good to me. Your reasoning to keep avoid going DRY at this point also makes sense to me. 
I also appreciate that you are keeping this PR small. Much easier to grok the test flow. 

I ran the tests locally and they all passed. I did get this ugly warning :
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```
but it didn't break anything. I'm guessing I have something off on my dev environment.

**Comment by ttahmouch** [comment]
> @ttahmouch The changes look good to me. Your reasoning to keep avoid going DRY at this point also makes sense to me. I also appreciate that you are keeping this PR small. Much easier to grok the test flow.

❤️ 🙏 Thank you so much for taking a look, @jhnstn . I appreciate it a lot. I'm glad you feel the same way regarding DRY.

> I ran the tests locally and they all passed. I did get this ugly warning :
> 
> ```
> Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
> ```
> 
> but it didn't break anything. I'm guessing I have something off on my dev environment.

I did get that ugly warning as well, but it didn't impact the test assertions since I saw them go from red to green when TDDing it. I thought about spending some time trying to fix whatever was wrong with the `forwardRef` in order to make the warning go away, but it didn't feel like something that needed to be included in this PR at the moment. Does that make sense?

**Comment by jhnstn** [comment]
> I did get that ugly warning as well, but it didn't impact the test assertions since I saw them go from red to green when TDDing it. I thought about spending some time trying to fix whatever was wrong with the forwardRef in order to make the warning go away, but it didn't feel like something that needed to be included in this PR at the moment. Does that make sense?

Agreed. Doesn't seem to block this work so no need to fix it here.

**Comment by ttahmouch** [comment]
All of the tests and workflows are ✅ now, @jhnstn . If you have some time soon, would you mind giving the remaining changes a cursory glance? They're fairly small.

**Review by jhnstn — APPROVED** [review]
The latest changes look good. Getting the snapshots sorted out does feel like a priority but your solution is the right choice for this effort

:shipit:

**Comment by ttahmouch** [comment]
> The latest changes look good. Getting the snapshots sorted out does feel like a priority but your solution is the right choice for this effort
> 
> :shipit:

I agree, and I appreciate your input and the review. It means a lot to me. I just wanted to time box the changes for this specific PR, and addressing the snapshot assertions was taking longer than I thought was reasonable at the moment. This just felt like a good compromise.

**Comment by hypest** [comment]
Yay, cool to see more automated tests merged! @ttahmouch , for easier reference, can you call out in the PR description exactly which tests (titles) have been added by the PR? And more interestingly, can we drop those from the manual testsuites or there are parts that are performed in manual but not yet in the automated tests so we need to keep those manual for now?

**Pull request merged**